### PR TITLE
Litecoin: enable support for SSE2 instructions in scrypt via configure --enable-sse2

### PR DESCRIPTION
**Pull request by romanornr**
SSE2 is missing for Litecoin.

Right now I have this implementation working for Viacoin.
https://github.com/viacoin/viacoin/tree/0.13-dev

Follow-up on my last pull request: https://github.com/litecoin-project/litecoin/pull/316

https://github.com/litecoin-project/litecoin/pull/316#issuecomment-305470698
modified scrypt_test.ccp to only run either SSE2 or non-SSE2 hashing 10000 times. On my low power Intel Atom J2900 CPU the SSE2 version is over **2x faster** (**42 vs 18 seconds**, just pasting one, the results are stable over multiple runs):
````# time ./test_litecoin --run_test=scrypt_tests
Running 1 test case...

*** No errors detected
41.942u 0.002s 0:42.06 99.7% 9063+838k 54+0io 532pf+0w

SSE2 enabled:
18.880u 0.008s 0:18.91 99.8% 9054+837k 2+0io 389pf+0w
````

To sum up:

- Significant performance boost SSE2 (2x faster)
- POW Validation is a  bottleneck, with SSE2 it can speed up the client.

### CODE REVIEW

**Comment by thrasher-** [comment]
ACK
